src/components/Filters/components/Input/Input.tsx

```diff
@@ -44,25 +49,37 @@ export const StyledInput = styled.input`
   :-ms-input-placeholder {
     color: ${getFormStyle('placeholderColor')};
   }
+
+  ${({ isInvalid }) =>
+    isInvalid &&
+    css`
+      ${stateStyles}
+      border-color: ${getFormStyle('invalidBorderColor')};
+    `}
 `;
 
 const Input: React.FC<InputProps> = ({
   value = '',
   onChange,
   maxLength,
-  pattern,
-  patternMessage,
+  pattern = patterns.string.pattern,
+  patternMessage = patterns.string.patternMessage,
+  validate = validateDefault,
   placeholder = 'String',
+  isDisabled = false,
 }) => {
-  const handleOnChange = (event) => {
-    if (patternMessage) {
-      const { target } = event;
-      target.setCustomValidity(
-        target.validity.patternMismatch ? patternMessage : '',
-      );
-    }
+  const [isInvalid, setIsInvalid] = useState(false);
 
+  const handleOnValidate = (event) => {
+    const invalid = onValidate(event, validate, patternMessage);
```

Comment:
It seems to me to be unnecessarily complicated. It will get the value, call validate, it will return good or bad and we will display error message and invalidate the apply button. Or I am wrong?